pkg/scheduler/framework/runtime/framework_test.go

```diff
@@ -0,0 +1,162 @@
+package runtime
+
+import (
+	"context"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	clusterv1alpha1 "github.com/karmada-io/karmada/pkg/apis/cluster/v1alpha1"
+	"github.com/karmada-io/karmada/pkg/scheduler/framework"
+	frameworktesting "github.com/karmada-io/karmada/pkg/scheduler/framework/testing"
+)
+
+func Test_frameworkImpl_RunFilterPlugins(t *testing.T) {
+	ctx := context.Background()
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	alwaysError := frameworktesting.NewMockFilterPlugin(mockCtrl)
+	alwaysError.EXPECT().Filter(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).AnyTimes().Return(framework.NewResult(framework.Error))
+
+	alwaysSuccess := frameworktesting.NewMockFilterPlugin(mockCtrl)
+	alwaysSuccess.EXPECT().Filter(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).AnyTimes().Return(framework.NewResult(framework.Success))
+
+	tests := []struct {
+		name      string
+		plugins   []framework.Plugin
+		isSuccess bool
+	}{
+		{
+			name:      "no filter plugin",
+			plugins:   []framework.Plugin{},
+			isSuccess: true,
+		},
+		{
+			name:      "error filter plugin",
+			plugins:   []framework.Plugin{alwaysError},
+			isSuccess: false,
+		},
+		{
+			name:      "success filter plugin",
+			plugins:   []framework.Plugin{alwaysSuccess},
+			isSuccess: true,
+		},
+		{
+			name:      "error and success filter plugins",
+			plugins:   []framework.Plugin{alwaysError, alwaysSuccess},
+			isSuccess: false,
+		},
+		{
+			name:      "success and error filter plugins",
+			plugins:   []framework.Plugin{alwaysSuccess, alwaysError},
+			isSuccess: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			registry, err := createAndRegisterFactory(tt.plugins)
+			if err != nil {
+				t.Errorf("create plugin factory error:%v", err)
+			}
+
+			frameWork, err := NewFramework(registry)
+			if err != nil {
+				t.Errorf("create frame work error:%v", err)
+			}
+
+			result := frameWork.RunFilterPlugins(ctx, nil, nil, nil)
+			if result.IsSuccess() != tt.isSuccess {
+				t.Errorf("want %v, but get:%v", tt.isSuccess, result.IsSuccess())
+			}
+		})
+	}
+}
+
+func createAndRegisterFactory(plugins []framework.Plugin) (Registry, error) {
+	registry := Registry{}
+	for _, plugin := range plugins {
+		filterPluginFactory := func() (framework.Plugin, error) {
+			return plugin, nil
+		}
+		if err := registry.Register("foo", filterPluginFactory); err != nil {
```

Comment:
we can not register plugins with the same name, they are override.

Reply:
Every time createAndRegisterFactory is called, the registry is created new, so there is no override.